### PR TITLE
Powerful setup.sh

### PR DESCRIPTION
**Pull request by simonsmh**
Short Description: 

Very powerful run.sh with lots of function.
1.install(make .so)
2.update
3.config generator
4.config backup
5.run loop make it never get down

It should run like 
run.sh *.json or other opinion. 
See -help.

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -1,26 +1,145 @@
...
+sed -i "s/GOOGLE_MAPS_API_KEY/$gmapkey/g" configs/config.json
+echo "edit configs/config.json to modify any other config."
+}
+
+function Pokebotinstall () {
+cd $pokebotpath
+echo "1.Debian/Ubuntu"
+echo "2.Centos/RedHat"
+echo "3.Mac os"
+echo "4.Other"
```
Comment (terbolous) [inline run.sh]: This can generally be probed with stuff like `lsb_release` or `uname`

File: run.sh [hunk 2]
```diff
@@ -1,26 +1,145 @@
-#!/usr/bin/env bash
+#!/bin/bash
+pokebotpath="/root/PokemonGo-Bot"
+backuppath="/root/PokemonGo-Bot/backup"
```
Comment (terbolous) [inline run.sh]: hard coded paths :-(

Comment (simonsmh) [inline run.sh]: it can be edited.
Let me fix it.

File: run.sh (hunk 1 above)
Comment (simonsmh) [inline run.sh]: I can't fix it because I don't have mac or centos devices. It's just working. Can you help me to fix it? Thx.

**Comment by solderzzc** [comment]
run.sh is a script to run configuration, we need separate setup to setup.sh.

**Comment by simonsmh** [comment]
@solderzzc Name it pokebot.sh is better.

**Comment by simonsmh** [comment]
Does it really need so much shell? :P

**Comment by solderzzc** [comment]
yes, I have two scripts running by ./run.sh, so we'd better keep it for other users.

File: run.sh [hunk 3]
```diff
@@ -1,26 +1,17 @@
 #!/usr/bin/env bash
-
-# Starts PokemonGo-Bot
-config=""
-
-if [ ! -z $1 ]; then
```
Comment (solderzzc) [inline run.sh]: this one is still needed for multiple configurations.

**Comment by simonsmh** [comment]
Is it OK? I edited run.sh setup.sh install.sh and removed setup.py which is not fit to use. The usage of run.sh is different but easier.

File: run.sh (hunk 3 above)
Comment (simonsmh) [inline run.sh]: OK I will revert it.

**Comment by simonsmh** [comment]
Maybe every new user here should use this new install.sh or setup.sh to make them use this bot in a very short time. The guide of how to install may should be updated too. :p

File: run.sh [hunk 4]
```diff
@@ -1,26 +1,21 @@
 #!/usr/bin/env bash
-
-# Starts PokemonGo-Bot
-config=""
-
+pokebotpath=$(pwd)
+filename=""
 if [ ! -z $1 ]; then
```
Comment (solderzzc) [inline run.sh]: This is needed.

**Comment by solderzzc** [comment]
great, waiting for the document :) we need add doc folder

**Pull request merged**